src/geometry.rs

```diff
@@ -432,9 +432,21 @@ impl TriangleFace {
             i1 != i2 && i1 != i3 && i2 != i3,
             "One or more face edges consists of the same vertex"
         );
-        TriangleFace {
-            vertices: (i1, i2, i3),
-            normals: (i1, i2, i3),
+        if i1 < i2 && i1 < i3 {
+            TriangleFace {
+                vertices: (i1, i2, i3),
+                normals: (i1, i2, i3),
+            }
+        } else if i2 < i1 && i2 < i3 {
+            TriangleFace {
+                vertices: (i2, i3, i1),
+                normals: (i2, i3, i1),
+            }
+        } else {
+            TriangleFace {
+                vertices: (i3, i1, i2),
+                normals: (i3, i1, i2),
+            }
```

Comment:
Wouldn't it be cleaner if you had array `[i1, i2, i3]`, sorted it and then moved elements 0, 1 and 2 into tuple?

Reply:
Nononono (unless I misunderstood you). This is not about sorting! I'm only finding the lowest index and from there on, the order remains unchanged.

Reply:
Carry on then.